### PR TITLE
Adding Python Implementation for Kosaraju Algorithm (Issue #2246)

### PR DESCRIPTION
**Pull request by sukritishah15**
Fixes #

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- 
- 

#### Languages Used:
- Python

#### Files Added:
- Kosaraju_Algorithm.py

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Kosaraju_Algorithm/Kosaraju_Algorithm.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+# Kosaraju’s Algorithm involves two passes of Depth First Search in order to find the number of SCCs
+
+# defaultdict is a useful technique for building graphs.
+from collections import defaultdict 
+
+# Function for first DFS pass and to fill vertices in finishStack
+def firstDFS(g, v, visited, finishStack): 
+	visited[v] = True
+	for i in g[v]: 
+		if visited[i] == False:
```
Comment (singh-shreya6) [inline Kosaraju_Algorithm/Kosaraju_Algorithm.py]: Too many spaces indention. Please give 4 space indentation at the start of each block.

File: Kosaraju_Algorithm/Kosaraju_Algorithm.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+		if visited[i] == False: 
+			firstDFS(g, i, visited, finishStack) 
+	finishStack = finishStack.append(v) 
+
+# Function for second DFS pass
+def secondDFS(gt, v, visited): 
+	visited[v] = True
+	print(v, end = " ")
+	for i in gt[v]: 
+		if visited[i] == False:
```
Comment (singh-shreya6) [inline Kosaraju_Algorithm/Kosaraju_Algorithm.py]: Indentation

File: Kosaraju_Algorithm/Kosaraju_Algorithm.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+v = int(input("Enter number of vertices in the graph: "))
+
+# g represents original graph 
+g = defaultdict(list)
+
+# gt represents transpose of original graph
+gt = defaultdict(list)
+
+e = int(input("Enter number of edges in the graph: "))
+for i in range(1,e+1):
```
Comment (singh-shreya6) [inline Kosaraju_Algorithm/Kosaraju_Algorithm.py]: Space around operator +. Also space after comma.

File: Kosaraju_Algorithm/Kosaraju_Algorithm.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+    num = list(map(int, input(' ').split()))
+    g[num[0]].append(num[1])
+    gt[num[1]].append(num[0])
+
+# Stack to be filled according to finishing time
+finishStack = []
+
+# Maintaining visited array of size v
+# All vertices are marked False for first DFS pass
+visited = [False]*v
```
Comment (singh-shreya6) [inline Kosaraju_Algorithm/Kosaraju_Algorithm.py]: Space around *

File: Kosaraju_Algorithm/Kosaraju_Algorithm.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+# First DFS on g
+for i in range(v): 
+	if visited[i] == False: 
+		firstDFS(g, i, visited, finishStack)
+	
+print ("Strongly connected components(SCCs) in the graph are: ") 
+	
+# Maintaining visited array of size v
+# All vertices are marked False for second DFS pass
+visited = [False]*v
```
Comment (singh-shreya6) [inline Kosaraju_Algorithm/Kosaraju_Algorithm.py]: Space around *

File: Kosaraju_Algorithm/Kosaraju_Algorithm.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+
+# Second DFS on gt
+while finishStack: 
+	i = finishStack.pop() 
+	if visited[i] == False: 
+		secondDFS(gt, i, visited)
```
Comment (singh-shreya6) [inline Kosaraju_Algorithm/Kosaraju_Algorithm.py]: Indentation error.

**Comment by sukritishah15** [comment]
Please re-review the PR.

**Pull request merged**